File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
public class Solution {
    public int[] solution(int[] sequence, int k) {
        int start = 0, end = 0;
        int sum = 0;
        int minLength = Integer.MAX_VALUE;
        int[] answer = new int[2];

        while (end < sequence.length) {
            sum += sequence[end];

            while (sum >= k) {
                if (sum == k && (end - start + 1) < minLength) {
                    minLength = end - start + 1;
                    answer[0] = start;
                    answer[1] = end;
                }
                sum -= sequence[start];
                start++;
            }
            end++;
        }
        return answer;
    }
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    n = len(sequence)
    start, end = 0,0
    total = sequence[0]
    res = [0,n-1]
    while start<n and end<n:
        if total==k:
            if (end-start)<(res[1]-res[0]):
                res=[start, end]
            start +=1
            if start<n:
                total -= sequence[start-1]
        elif total<k:
            end+=1
            if end<n:
                total += sequence[end]
        else:
            total -= sequence[start]
            start+=1
    return res